appinventor/appengine/src/com/google/appinventor/client/explorer/SourceStructureExplorer.java

```diff
@@ -68,10 +68,14 @@ public void onBrowserEvent(Event event) {
     }
   }
 
+  public SourceStructureExplorer() {
+    this(true);
+  }
+
   /**
    * Creates a new source structure explorer.
    */
-  public SourceStructureExplorer() {
+  public SourceStructureExplorer(boolean includeButtonPanel) {
```

Comment:
You restored the includeButtonPanel parameter, which is commented out in the code. Is there some reason to keep it?

Reply:
So, ideally, those buttons should be located somewhere else, not at the bottom of that panel. That's why the conditional rendering, even though now we are always adding them as we don't have any other location.

Reply:
> So, ideally, those buttons should be located somewhere else, not at the bottom of that panel. That's why the conditional rendering, even though now we are always adding them as we don't have any other location.

Hmm. If we allow the buttons to move in the future, it will probably be done a different way using UIBinder. I don't think leaving in the unused boolean will save us any time if we add this functionality later.

Reply:
Okay, I'll get rid of it